tests/Unit/Elliptic/Actions/Test_InitializeBackgroundFields.cpp

```diff
@@ -0,0 +1,107 @@
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Creators/Interval.hpp"
+#include "Domain/Creators/RegisterDerivedWithCharm.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Actions/InitializeBackgroundFields.hpp"
+#include "Elliptic/Tags.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "ParallelAlgorithms/DiscontinuousGalerkin/InitializeDomain.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+
+struct BackgroundFieldTag : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct System {
+  using background_fields = tmpl::list<BackgroundFieldTag>;
+};
+
+struct Background {
+  static tuples::TaggedTuple<BackgroundFieldTag> variables(
+      const tnsr::I<DataVector, 1>& x,
+      tmpl::list<BackgroundFieldTag> /*meta*/) noexcept {
+    return {Scalar<DataVector>{get<0>(x)}};
+  }
+  // NOLINTNEXTLINE
+  void pup(PUP::er& /*p*/) noexcept {}
+};
+
+template <typename Metavariables>
+struct ElementArray {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementId<1>;
+  using const_global_cache_tags = tmpl::list<domain::Tags::Domain<1>>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+                         tmpl::list<domain::Tags::InitialRefinementLevels<1>,
+                                    domain::Tags::InitialExtents<1>>>,
+                     Actions::SetupDataBox, dg::Actions::InitializeDomain<1>>>,
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<::elliptic::Actions::InitializeBackgroundFields<
+              typename Metavariables::system,
+              elliptic::Tags::Background<Background>>>>>;
+};
+
+struct Metavariables {
+  using system = System;
+  using component_list = tmpl::list<ElementArray<Metavariables>>;
+  using const_global_cache_tags =
+      tmpl::list<elliptic::Tags::Background<Background>>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Elliptic.Actions.InitializeBackgroundFields",
+                  "[Unit][Elliptic][Actions]") {
+  domain::creators::register_derived_with_charm();
+  // Which element we work with does not matter for this test
+  const ElementId<1> element_id{0, {{SegmentId{2, 1}}}};
+  const domain::creators::Interval domain_creator{{{-0.5}}, {{1.5}},   {{2}},
+                                                  {{4}},    {{false}}, nullptr};
```

Comment:
Very strange formatting above... is this clang-format trying to align arguments? 😕 If it is what clang-format wants, then so be it... no change needed

Reply:
yep this is clang-format's doing. I think it's not too bad, it just aligns the arguments in a grid 🤷‍♂️